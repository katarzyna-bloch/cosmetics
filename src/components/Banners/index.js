import PropTypes from 'prop-types'

import Banner from '../Banner'
import Carousel from '../Carousel'

const Banners = ({ banners }) => {
  const items = banners.map(banner => (
    <Banner banner={banner}>{banner.src}</Banner>
  ))
  
  return (
    <Carousel
      items={items}
      settings={{ slidesToShow: 1, autoplay: true, dots: true }}
    />
  )
}

Banners.propTypes = {
  banners: PropTypes.array.isRequired,
}

export default Banners
