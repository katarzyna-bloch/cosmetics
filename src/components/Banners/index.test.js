import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Banners from '.'

jest.mock('../../components/Carousel', () => {
  const Carousel = () => null
  return Carousel
})

const props = {
  banners: [
    { id: '1', img: 'pielegnacja.jpg', url: '/categories/care', text: 'Miks produktów aby zadbać o swoją cere.' },
    { id: '2', img: 'perfumy.jpg', url: '/categories/perfume', text: 'Bestsellery perfum w atrakcyjnych cenach.' },
    { id: '3', img: 'makeup.jpg', url: '/categories/makeup', text: 'Różnorodne rabaty dla Ciebie.' },
  ]
}

it('should render correctly', () => {
  const wrapped = shallow(<Banners {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
