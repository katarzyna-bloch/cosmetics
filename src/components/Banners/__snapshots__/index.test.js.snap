// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Carousel
  items={
    Array [
      <Banner
        banner={
          Object {
            "id": "1",
            "img": "pielegnacja.jpg",
            "text": "Miks produktów aby zadbać o swoją cere.",
            "url": "/categories/care",
          }
        }
      />,
      <Banner
        banner={
          Object {
            "id": "2",
            "img": "perfumy.jpg",
            "text": "Bestsellery perfum w atrakcyjnych cenach.",
            "url": "/categories/perfume",
          }
        }
      />,
      <Banner
        banner={
          Object {
            "id": "3",
            "img": "makeup.jpg",
            "text": "Różnorodne rabaty dla Ciebie.",
            "url": "/categories/makeup",
          }
        }
      />,
    ]
  }
  settings={
    Object {
      "autoplay": true,
      "dots": true,
      "slidesToShow": 1,
    }
  }
/>
`;
